### PR TITLE
25 mar

### PR DESCRIPTION
**Pull request by MarianBastiurea**

### CODE REVIEW

File: src/main/java/org/example/c17inputoutput/ReadingAFileChallenge.java [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+
+public class ReadingAFileChallenge {
+    public static void main(String[] args) {
+
+        try (BufferedReader br = new BufferedReader(
+                new FileReader("testfile.txt"))) {
+
+//            System.out.printf(" %,d lines in file%n", br.lines().count());
+
+            Pattern pattern = Pattern.compile("\\p{javaWhitespace}+");
```
Comment (andreiursudev) [inline src/main/java/org/example/c17inputoutput/ReadingAFileChallenge.java]: Give this Pattern object a suggestive name.

File: src/main/java/org/example/c17inputoutput/ReadingAFileChallenge.java [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+            e.printStackTrace();
+        }
+
+        System.out.println("--------------------");
+        String input = null;
+        try {
+            input = Files.readString(Path.of("testfile.txt"));
+            input = input.replaceAll("\\p{Punct}", "");
+
+            Pattern pattern = Pattern.compile("\\w{5,}");
```
Comment (andreiursudev) [inline src/main/java/org/example/c17inputoutput/ReadingAFileChallenge.java]: Give this Pattern object a suggestive name.

File: src/main/java/org/example/c17inputoutput/ReadingAFileWithNIO2.java [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+        System.out.println(System.getProperty("file.encoding"));
+        System.out.println(Charset.defaultCharset());
+
+        Path path = Path.of("pathfile.txt");
+        try {
+            System.out.println(new String(Files.readAllBytes(path)));
+            System.out.println("----------------");
+            System.out.println(Files.readString(path));
+
+            Pattern p = Pattern.compile("(.{15})(.{3})(.{12})(.{8})(.{2}).*");
```
Comment (andreiursudev) [inline src/main/java/org/example/c17inputoutput/ReadingAFileWithNIO2.java]: Give this Pattern object a suggestive name.

**Review by andreiursudev — APPROVED** [review]

File: src/main/java/org/example/c17inputoutput/ReadingAFileChallenge.java (hunk 1 above)
Comment (MarianBastiurea) [reply src/main/java/org/example/c17inputoutput/ReadingAFileChallenge.java]: done. New name is splitToWords

File: src/main/java/org/example/c17inputoutput/ReadingAFileChallenge.java (hunk 2 above)
Comment (MarianBastiurea) [reply src/main/java/org/example/c17inputoutput/ReadingAFileChallenge.java]: Done. New name splitInWordsWithMoreThan5Char

File: src/main/java/org/example/c17inputoutput/ReadingAFileWithNIO2.java (hunk 3 above)
Comment (MarianBastiurea) [reply src/main/java/org/example/c17inputoutput/ReadingAFileWithNIO2.java]: Done. New name is giveDimensionForColumn

File: src/main/java/org/example/c17inputoutput/ReadingAFileChallenge.java (hunk 1 above)
Comment (andreiursudev) [reply src/main/java/org/example/c17inputoutput/ReadingAFileChallenge.java]: Not good. Name should describe the pattern, not how the pattern is used. This pattern can be used for many things not only for splitting words. You can call this pattern is for `oneOrManyWhitespaces`.

File: src/main/java/org/example/c17inputoutput/ReadingAFileChallenge.java (hunk 2 above)
Comment (andreiursudev) [reply src/main/java/org/example/c17inputoutput/ReadingAFileChallenge.java]: Same comment as above.

File: src/main/java/org/example/c17inputoutput/ReadingAFileWithNIO2.java (hunk 3 above)
Comment (andreiursudev) [reply src/main/java/org/example/c17inputoutput/ReadingAFileWithNIO2.java]: Same comment as above.

File: src/main/java/org/example/c17inputoutput/ReadingAFileChallenge.java (hunk 1 above)
Comment (MarianBastiurea) [reply src/main/java/org/example/c17inputoutput/ReadingAFileChallenge.java]: ok. change it

File: src/main/java/org/example/c17inputoutput/ReadingAFileChallenge.java (hunk 2 above)
Comment (MarianBastiurea) [reply src/main/java/org/example/c17inputoutput/ReadingAFileChallenge.java]: moreThan5Char

File: src/main/java/org/example/c17inputoutput/ReadingAFileWithNIO2.java (hunk 3 above)
Comment (MarianBastiurea) [reply src/main/java/org/example/c17inputoutput/ReadingAFileWithNIO2.java]: groupsChar

File: src/main/java/org/example/c17inputoutput/ReadingAFileChallenge.java (hunk 2 above)
Comment (andreiursudev) [reply src/main/java/org/example/c17inputoutput/ReadingAFileChallenge.java]: `\w`means any word character(letter, number, underscore). Char means any character which includes more than word characters.

**Pull request merged**